Brokerages/Bitfinex/BitfinexBrokerage.Util.cs

```diff
@@ -0,0 +1,164 @@
+            { new OrderTypeMap { BitfinexOrderType = "stop", Wallet = WalletType.trading.ToString(), OrderType = OrderType.StopMarket } },
+            //{ new OrderTypeMap { BitfinexOrderType = "trailing stop", Wallet = WalletType.trading.ToString(), OrderType = OrderType.StopLimit } },
+        };
+        #endregion
+
+
+        /// <summary>
+        /// Map exchange status
+        /// </summary>
+        /// <param name="response"></param>
+        /// <returns></returns>
+        public static OrderStatus MapOrderStatus(BitfinexOrderStatusResponse response)
+        {
+            decimal remainingAmount;
+            decimal executedAmount;
+            if (response.IsCancelled)
+            {
+                return OrderStatus.Canceled;
+            }
+            else if (decimal.TryParse(response.RemainingAmount, out remainingAmount) && remainingAmount > 0
+                && decimal.TryParse(response.ExecutedAmount, out executedAmount) && executedAmount > 0)
+            {
+                return OrderStatus.PartiallyFilled;
+            }
+            else if (response.IsLive)
+            {
+                return OrderStatus.Submitted;
+            }
+
+            return OrderStatus.Invalid;
+        }
+
+
+        /// <summary>
+        /// Map exchange order type
+        /// </summary>
+        /// <param name="orderType"></param>
+        /// <returns></returns>
+        public string MapOrderType(OrderType orderType)
+        {
+            var result = _orderTypeMap.Where(o => o.Wallet == Wallet && o.OrderType == orderType);
+
+            if (result != null & result.Count() == 1)
+            {
+                return result.Single().BitfinexOrderType;
+            }
+
+            throw new Exception("Order type not supported: " + orderType.ToString());
+        }
+
+
+        /// <summary>
+        /// Map exchange order type
+        /// </summary>
+        /// <param name="orderType"></param>
+        /// <returns></returns>
+        public OrderType MapOrderType(string orderType)
+        {
+            var result = _orderTypeMap.Where(o => o.Wallet == Wallet && o.BitfinexOrderType == orderType);
+
+            if (result != null & result.Count() == 1)
+            {
+                return result.Single().OrderType;
+            }
+
+            throw new Exception("Order type not supported: " + orderType.ToString());
+        }
+
+        /// <summary>
+        /// Map exchange order status
+        /// </summary>
+        /// <param name="msg"></param>
+        /// <returns></returns>
+        public OrderStatus MapOrderStatus(TradeMessage msg)
+        {
+            if (Math.Abs(msg.FEE) != 0)
+            {
+                var cached = this.CachedOrderIDs.Where(c => c.Value.BrokerId.Contains(msg.TRD_ORD_ID.ToString())).FirstOrDefault();
+                if (cached.Value != null)
+                {
+                    if (msg.TRD_ORD_ID == cached.Value.BrokerId.Select(b => int.Parse(b)).Max(b => b))
```

Comment:
We are looking for the Lean order Id that corresponds to a broker id that accompanies a trade fill. If these don't states don't align, I think the Max should throw an exception.

I've removed the redundancy in the Max call [here](https://github.com/jameschch/Lean/commit/ab66b36e0077eb6e5aad6feb06eb2c03c4c554e2)